src/services/Login.ts

```diff
@@ -0,0 +1,154 @@
+import * as API from './Api';
+
+import { AuthInfo, AuthStrategy, AuthResult } from '../types/Auth';
+import { LoginSession } from '../store/Store';
+import { TimeInMilliseconds } from '../types/Common';
+
+// We just mock the types for state for now, to avoid injecting more
+// dependencies on the service.
+type Dispatch = any;
+type AppState = any;
+
+export interface LoginResult {
+  status: AuthResult;
+  session?: LoginSession;
+  error?: any;
+}
+
+interface LoginStrategy<T = {}> {
+  prepare: (info: AuthInfo) => Promise<AuthResult>;
+  perform: (request: DispatchRequest<T>) => Promise<LoginResult>;
+}
+
+interface DispatchRequest<T> {
+  dispatch: Dispatch;
+  getState: () => AppState;
```

Comment:
Why is this a function and not a state? We want to pass the state, not the method that calls returns the state. So wouldn't this be `KialiAppState` as returned by `mapStateToProps`?